### ISSUE TITLE
how can I get pose.mat?

### ISSUE THREAD

**Issue by chdzhen** [yilundu/cross_attention_renderer#11]
'''
    def create_dataloader_callback(sidelength, batch_size, query_sparsity):
        train_dataset = RealEstate10k(img_root="data_download/realestate/train",
                                     pose_root="poses/realestate/train.mat",
                                     num_ctxt_views=opt.views, num_query_views=1, query_sparsity=192,
                                     lpips=opt.lpips, augment=(not opt.no_data_aug))

        .....
        val_dataset = RealEstate10k(img_root="data_download/realestate/test",
                                      pose_root="poses/realestate/test.mat",
                                      num_ctxt_views=opt.views, num_query_views=1, augment=False)
'''
pose_root="poses/realestate/train.mat",
pose_root="poses/realestate/test.mat",

**Comment by yilundu** [comment]
You can find it here: https://www.dropbox.com/s/qo8b7odsms722kq/cvpr2023_wide_baseline_data.tar.gz?dl=0

**Comment by chdzhen** [comment]
thanks!
can you share the generate *.mat script?

**Comment by yilundu** [comment]
Yes it should be below:



```
import glob
from tqdm import tqdm
import numpy as np
from scipy.io import savemat


if __name__ == "__main__":
    files = glob.glob("train/*.txt")

    full_dict = {}

    for file in tqdm(files):
        # file_dict = {}
        data = np.loadtxt(file, skiprows=1)

        # if len(data.shape) == 1:
        #     continue

        # name, entry = data[:, 0], data[:, 1:]

        # for n, e in zip(name, entry):
        #     file_dict[n] = e

        base = file.split("/")[-1].split(".")[0]
        full_dict[base] = data

    savemat("train.mat", full_dict, do_compression=True)
    import pdb
    pdb.set_trace()
    print(files)
```

**Comment by chdzhen** [comment]
why set HW to H, W = 256, 455?
![Uploading 截屏2024-10-23 17.08.53.png…]()

**Comment by chdzhen** [comment]
why HW set to 256, 455?

in realestate10k_dataio.py   

    class RealEstate10K():
        def init():
        ***
          data = np.load(dummy_img_path)
          key = list(data.keys())[0]
          im = data[key]
          print(im.shape)
          H, W = im.shape[:2]
          H, W = 256, 455
          self.H, self.W = H, W
          self.augment = augment
  
          self.square_crop = square_crop
  
          xscale = W / min(H, W)
          yscale = H / min(H, W)
  
          dim = min(H, W)
  
          self.xscale = xscale
          self.yscale = yscale
  
          # For now the images are already square cropped
          self.H = 256
          self.W = 455
        ***
